src/xlsx/conversion/table_conversion.ts

```diff
@@ -199,70 +200,133 @@ function convertTableFormulaReferences(convertedSheets: SheetData[], xlsxSheets:
 }
 
 /**
- * Convert table-specific references in formulas into standard references.
+ * Convert table-specific references in formulas into standard references. A table reference is composed of columns names,
+ * and of keywords determining the rows of the table to reference.
  *
  * A reference in a table can have the form (only the part between brackets should be given to this function):
  *  - tableName[colName] : reference to the whole column "colName"
+ *  - tableName[#keyword] : reference to the whatever row the keyword refers to
  *  - tableName[[#keyword], [colName]] : reference to some of the element(s) of the column colName
+ *  - tableName[[#keyword], [colName]:[col2Name]] : reference to some of the element(s) of the columns colName to col2Name
+ *  - tableName[[#keyword1], [#keyword2], [colName]] : reference to all the rows referenced by the keywords in the column colName
+ *  - tableName[[#keyword1], [colName], [#keyword2]]: the keywords and colName can be in any order
+ *
  *
  * The available keywords are :
  * - #All : all the column (including totals)
  * - #Data : only the column data (no headers/totals)
  * - #Headers : only the header of the column
  * - #Totals : only the totals of the column
  * - #This Row : only the element in the same row as the cell
+ *
+ * Note that the only valid combination of multiple keywords are #Data + #Totals and #Headers + #Data.
  */
-function convertTableReference(expr: string, table: XLSXTable, cellXc: string) {
-  const refElements = expr.split(",");
+function convertTableReference(
+  sheetPrefix: string,
+  expr: string,
+  table: XLSXTable,
+  cellXc: string
+) {
+  // TODO: Ideally we'd want to make a real tokenizer, this simple approach won't work if for example the column name
```

Comment:
why not make a real tokenizer and parser then ? The code would probably be more readable and maintainable. It's probably not that complex compared to this code which is not easy either.

Also, we might implement this syntax in the future (even though it raises a few questions), so we might as well write the parsing code nicely the first time :)